### ISSUE TITLE
replace zxing with https://github.com/nayuki/QR-Code-generator 

### ISSUE THREAD

**Issue by syjer** [alfio-event/alf.io#698]
As we generate only the qrcode, we don't need all the part that read them.

https://github.com/nayuki/QR-Code-generator

**Comment by omsitapara23** [comment]
I would like to do this job if anyone is not doing it already, also can you guide me from which file should I start

**Comment by syjer** [comment]
hi @omsitapara23 , thank you for the interest. Currently nobody is working on this issue, so feel free to try :).

There is exactly one file that you need to modify : https://github.com/alfio-event/alf.io/blob/master/src/main/java/alfio/util/ImageUtil.java

Note: I've noticed that the library that I've linked, does not support like zxing setting a size (in this case, 200px * 200px:

`MultiFormatWriter().encode(text, BarcodeFormat.QR_CODE, 200, 200, hintMap)`
).

You will need to check if it's possible to do the conversion.

**Comment by omsitapara23** [comment]
The pixels qr code generated using [https://github.com/nayuki/QR-Code-generator](url) depends on the version selected not the manual input. So I don't think we would be able to change the current implementation @syjer.

Thanks for the opportunity tho !! at least i learned how QR codes are formed :-)

**Comment by cbellone** [comment]
Hi @omsitapara23,

thank you very much for your analysis!
Can you please share more details about your findings?

Maybe we could resize the QR Code image _after_ generating it... if the cost of doing it would be reasonable.

Thank you!
Celestino

**Comment by omsitapara23** [comment]
Hi @cbellone,

Here are the insights of my findings:
* The function encodeText takes a string which needs to be encoded and the Error correction level.
* Based on the Ecc and string to encode length they calculate the version of the QR code. Verison basically defines the size of the QR code ( how many dimensions of QR code needs to be generated to encode the current text with the given ECC )
* Once the version is defined they generate the QR code in their QR code object.
* Then the function toImage( scale, border ) is called to convert the generated QR code to BufferedImage object. The border is the padding of white which is generated   around the QR code. The Image generated has the dimensions of ( (size + border*2   )*scale, (size + border*2)*scale ) where size = version*4 + 17
 
 => for our code we need a BitMatrix to be returned so what we can do is generate the
bufferedImage and resize it and then convert it to BitMatrix

Also please let me know if I should try the above-discussed strategy.

Thank you!
Om Sitapara.

**Comment by omsitapara23** [comment]
Hi @cbellone,

Can you please have a look at the code _https://drive.google.com/file/d/1KRuX3090xpOm3h6A5dxfMtNGPkIGP6i0/view?usp=sharing_ where I have tried to encode a random string using both **zxing**  and new QR encoder comparing the time required by each method. The new QR code generator is quite slow compared to zxing.

The time is taken by new QR code generator = 168 ms
Time is taken by zxing = 27 ms

**Comment by syjer** [comment]
hi @omsitapara23 , thank you very much for the in depth investigation 👍 

With your data in hand, we can say that my initial idea was not that good :).

So I'll close this issue. Thank you very much 👍

**Comment by syjer** [comment]
hi @nayuki , thank you for taking your time for answering, we will do another try :)

**Comment by RushikeshSawant** [comment]
@syjer @cbellone I would like to give it a try with an alternate faster version suggested by @nayuki . Can you please assign this to me?

**Comment by cbellone** [comment]
Hi @RushikeshSawant ,
I've assigned the issue to you. 
Thanks, and looking forward to your results!

**Comment by nayuki** [comment]
Cool. It looks like RushikeshSawant wants to use my fast Java library.

I'd like to emphasize that according to the benchmark timings on https://www.nayuki.io/page/fast-qr-code-generator-library , generating a QR Code with my slow or fast library will take under 10 milliseconds. If you are getting timings like 100 ms, then I think your code hasn't been JIT-compiled or the time is spent on image resizing or something.

In other words, I expect that my [main (slow) library](https://github.com/nayuki/QR-Code-generator) should be perfectly suitable. It has a shorter, clearer codebase, and has Maven packaging.
